snapshot/src/test/java/io/zeebe/snapshots/impl/FileBasedReceivedSnapshotTest.java

```diff
@@ -148,7 +148,7 @@ public void shouldNotRemovePendingSnapshotOnCommittingSnapshotWhenHigher() throw
     assertThat(receiverPendingDir)
         .as(
             "the latest pending snapshot should not be deleted because it is newer than the persisted one")
-        .isDirectoryContaining(p -> p.equals(receivedSnapshot.getPath()));
+        .isDirectoryNotContaining(p -> !p.equals(receivedSnapshot.getPath()));
```

Comment:
Here we should use `isDirectoryContaining` because we want to check if the directory exists.

Reply:
But we also want to guarantee that it's the only one, no? I've introduced a `isDirectoryContainingExactly` in the next PR and already refactored. I can change it now, and at any rate it'll be improved in the next PR to check that it's exactly only the snapshot.

Reply:
In this test, it is more important to test the directory exists. With `isDirectoryNotContaining` we are not checking that.

Reply:
:+1: If it is improved in the next PR